### PR TITLE
Build: allow dynamically change docker image

### LINKED ISSUE #17891: Update the blocks of devenv to support several sql database versions
**What would you like to be added**:
Currently the `Dockerfile` and `docker-compose.yaml` refer to a specific database version.
It would be nice to make use of environmental variables in order to support several versions.

### PR DESCRIPTION
**Pull request by markelog**
Right now, only for couple DBs.
With the 2.1, you can do `${TAG:-5.6}` without needing the `.env` file, but
it seems it would be better for back-compat to support `.env` right now.

Fixes #17891

### CODE REVIEW

**Comment by marefr** [comment]
@markelog as you can see in issue #17891 @papagian is already working on this. Please sync with her how to merge your work.

**Comment by papagian** [comment]
@marefr my implementation for #17891 is almost similar to this one, but it didn't have time to test it throughly and therefore I hadn't create a PR. So I guess we can keep @markelog's.
I will review it ASAP!

File: Makefile [hunk 1]
```diff
@@ -125,7 +125,7 @@ devenv: devenv-down
 
 	@cd devenv; \
 	./create_docker_compose.sh $(targets) || \
-	(rm -rf docker-compose.yaml; exit 1)
+	(rm -rf {docker-compose.yaml,conf.tmp,.env}; exit 1)
```
Comment (papagian) [inline Makefile]: Is `conf.tmp` related with this issue? If not, I would proposed to be removed from this pull request and create a separate pull request for this one.

**Review by papagian — CHANGES_REQUESTED** [review]
Nice!
Maybe you could add somewhere some guidelines for overriding the default versions (5.6 for mysql and 9.3) by passing the `TAG` to the Makefile for example:
`make devenv sources=mysql TAG=5.7.18` & `make devenv sources=postgres TAG=9.2`

File: Makefile (hunk 1 above)
Comment (markelog) [reply Makefile]: Yeah, it is related, some of the blocks do use it and `create_docker_compose.sh` utilize it as well. 

I didn't know about it before tho

**Comment by markelog** [comment]
> Nice!
Maybe you could add somewhere some guidelines for overriding the default versions (5.6 for mysql and 9.3) by passing the TAG to the Makefile for example:
make devenv sources=mysql TAG=5.7.18 & make devenv sources=postgres TAG=9.2

Yeah, I have separate issue about that - https://github.com/grafana/grafana/issues/16107. In that document we should definitely mention the `TAG` thing

**Comment by markelog** [comment]
It turns out we already some docs for #16107 so I just mentioned them in there. Also rewrote the thing, so there wouldn't be any name collisions and naming is more explicit now

**Review by torkelo — APPROVED** [review]
Looks great
